crates/wasmtime/src/runtime/externals/table.rs

```diff
@@ -270,12 +283,31 @@ impl Table {
     pub fn grow(&self, mut store: impl AsContextMut, delta: u64, init: Ref) -> Result<u64> {
         let store = store.as_context_mut().0;
         let ty = self.ty(&store);
-        let init = init.into_table_element(store, ty.element())?;
         let (table, _gc_store) = self.wasmtime_table(store, iter::empty());
         // FIXME(#11179) shouldn't need to subvert the borrow checker
         let table: *mut _ = table;
         unsafe {
-            match (*table).grow(delta, init, store)? {
+            let result = match element_type(&ty) {
+                TableElementType::Func => {
+                    let element = init.into_table_func(store, ty.element())?;
+                    (*table).grow_func(store, delta, element)?
+                }
+                TableElementType::GcRef => {
+                    // FIXME: `grow_gc_ref` shouldn't require the whole store
+                    // and should require `AutoAssertNoGc`. For now though we
+                    // know that table growth doesn't trigger GC so it should be
+                    // ok to create a copy of the GC reference even though it's
+                    // not tracked anywhere.
```

Comment:
Why don't we make `grow_gc_ref` take an `AutoAssertNoGc`?

Reply:
That can't work right now as growth requires the whole `dyn VMStore` to perform a fiber suspension but `AutoAssertNoGc` only requires the `StoreOpaque`. Soon though...